### ISSUE TITLE
mitmdump v3.0.4 doesn't accept filter for positional argument

### ISSUE THREAD

**Issue by kikuyan** [mitmproxy/mitmproxy#3051]
##### Steps to reproduce the problem:

1. mitmdump '~u google\.com'

and I get:
Traceback (most recent call last):
  File "release/specs/mitmdump", line 3, in <module>
  File "mitmproxy/tools/main.py", line 155, in mitmdump
  File "mitmproxy/tools/main.py", line 116, in run
  File "mitmproxy/optmanager.py", line 212, in update
KeyError: 'Unknown options: view_filter'
[5149] Failed to execute script mitmdump

##### Any other comments? What have you tried so far?

mitmdump --set dumper_filter='~u google\.com'
works, but not so handy like v3.0.3 or before.

Maybe help text needs update, too. It is saying view_filter option.
mitmdump --help
usage: mitmdump [options] [filter]
positional arguments:
  filter_args           Filter expression, equivalent to setting both the
                        view_filter and save_stream_filter options.

##### System information

mitmdump --version
Mitmproxy: 3.0.4 binary
Python:    3.6.4
OpenSSL:   OpenSSL 1.1.0h  27 Mar 2018
Platform:  Darwin-16.7.0-x86_64-i386-64bit

**Comment by madsmtm** [comment]
Can confirm, on MacOS High Sierra, Windows 10 and Kali Linux (The only linux distribution i had lying around ;) )

```
Mitmproxy: 3.0.4 
Python:    3.6.0
OpenSSL:   OpenSSL 1.1.0h  27 Mar 2018
Platform:  Darwin-17.4.0-x86_64-i386-64bit
```

```
Mitmproxy: 3.0.4
Python:    3.6.5
OpenSSL:   OpenSSL 1.1.0h  27 Mar 2018
Platform:  Windows-10-10.0.16299-SP0
```

```
Mitmproxy: 3.0.4 
Python:    3.6.5rc1
OpenSSL:   OpenSSL 1.1.0h  27 Mar 2018
Platform:  Linux-4.13.0-kali1-amd64-x86_64-with-Kali-kali-rolling-kali-rolling
```

**Comment by cortesi** [comment]
Hi there. Thanks for the report - this slipped through the cracks in addon option refactoring. Now that our core has been decomposed, we're forced to be super precise about what exactly we mean with the default filter specification to mitmdump. which is a good thing. 

* Filtering when we read a file using the `-r` flag is simple. We set the `readfile_filter` option, each flow loaded from disk is checked, and is skipped if it doesn't match. I've fixed this in #3056. 
* When we are streaming flows live, things are trickier. At first glance, the user might expect mitmdump to kill flows that don't match the filter. However, a flow can match the filter specification at any point in its lifecycle as data is added progressively (clientconnect -> requestheaders -> request, etc.). This means we can only conclusively say that a flow *doesn't* match a filter after its last event, by which point it has already been sent and received. This doesn't fly. I think the only sane option here is to make clear this is a *display filter only*, which means we only set the `dumper_filter` option.
* The final case is writing flows to file with the `-w` flag. We need to set the `save_stream_filter` option to cover this. 

In all, this means the default filter is a shortcut for setting the `readfile_filter`, `dumper_filter` and `save_stream_filter` options.
